braindecode/models/msvtnet.py

```diff
@@ -0,0 +1,372 @@
+        self.d_model = d_model
+        self.pe = nn.Parameter(torch.zeros(1, seq_length, d_model))
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = x + self.pe
+        return x
+
+
+class _Transformer(nn.Module):
+    """
+    Transformer encoder module with learnable class token and positional encoding.
+
+    Parameters
+    ----------
+    seq_length : int
+        Sequence length of the input.
+    d_model : int
+        Dimensionality of the model.
+    num_heads : int
+        Number of heads in the multihead attention.
+    feedforward_ratio : float
+        Ratio to compute the dimension of the feedforward network.
+    drop_prob : float, optional
+        Dropout probability, by default 0.5.
+    num_layers : int, optional
+        Number of transformer encoder layers, by default 4.
+    """
+
+    def __init__(
+        self,
+        seq_length: int,
+        d_model: int,
+        num_heads: int,
+        feedforward_ratio: float,
+        drop_prob: float = 0.5,
+        num_layers: int = 4,
+    ) -> None:
+        super().__init__()
+        self.cls_embedding = nn.Parameter(torch.zeros(1, 1, d_model))
+        self.pos_embedding = _PositionalEncoding(seq_length + 1, d_model)
+
+        dim_ff = int(d_model * feedforward_ratio)
+        self.dropout = nn.Dropout(drop_prob)
+        self.trans = nn.TransformerEncoder(
+            nn.TransformerEncoderLayer(
+                d_model,
+                num_heads,
+                dim_ff,
+                drop_prob,
+                batch_first=True,
+                norm_first=True,
+            ),
+            num_layers,
+            norm=nn.LayerNorm(d_model),
+        )
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        batch_size = x.shape[0]
+        x = torch.cat((self.cls_embedding.expand(batch_size, -1, -1), x), dim=1)
+        x = self.pos_embedding(x)
+        x = self.dropout(x)
+        return self.trans(x)[:, 0]
+
+
+class _DenseLayers(nn.Sequential):
+    """
+    Final classification layers.
+
+    Parameters
+    ----------
+    linear_in : int
+        Input dimension to the linear layer.
+    n_classes : int
+        Number of output classes.
+    """
+
+    def __init__(self, linear_in: int, n_classes: int):
+        super().__init__(
+            nn.Flatten(),
+            nn.Linear(linear_in, n_classes),
+        )
+
+
+class MSVTNet(EEGModuleMixin, nn.Module):
+    """MSVTNet model from Liu K et al (2024) from [msvt2024]_.
+
+    This model implements a multi-scale convolutional transformer network
+    for EEG signal classification, as described in [msvt2024]_.
+
+    .. figure:: https://raw.githubusercontent.com/SheepTAO/MSVTNet/refs/heads/main/MSVTNet_Arch.png
+       :align: center
+       :alt: MSVTNet Architecture
+
+    Parameters
+    ----------
+    n_filters_list : List[int], optional
+        List of filter numbers for each TSConv block, by default [9, 9, 9, 9].
+    conv1_kernel_sizes : List[int], optional
```

Comment:
```suggestion
    conv1_kernels_size : List[int], optional
```
Nice to add `s` to mark plural and to indicate it is a list of kernel size. I think it is better to put the `s` on kernel.